### DISCUSSION TITLE
Failed trying out samples in installation guide

### DISCUSSION THREAD

**Discussion by zhanggiene** [boostorg/ublas#171] [Q&A] [answered]
i was following the ublas instalation guidance and tried the example and test without changing anything. 
content from user-config.jam:   using gcc : : gcc-11 : <cxxflags>"-std=c++20" <linkflags>"" <link>"-lboost_unit_test_framework" ;
Operating system : Macos, 
compiler: gcc (Homebrew GCC 11.2.0_3) 11.2.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

command : 
../../../b2 -j8 release toolset=gcc cxxflags="-O3 -std=c++20" examples/tensor

output:
Performing configuration checks

    - default address-model    : 64-bit (cached) [1]
    - default architecture     : x86 (cached) [1]
    - symlinks supported       : yes (cached)
    - cxx17_if_constexpr       : yes (cached) [2]
    - cxx17_if_constexpr       : yes (cached) [2]

[1] gcc-11
[2] gcc-11/release/python-2.7/threadapi-pthread/threading-multi/visibility-hidden
...patience...
...patience...
...found 2766 targets...
...updating 4 targets...
gcc.compile.c++ ../../../bin.v2/libs/numeric/ublas/examples/tensor/gcc-11/release/threading-multi/visibility-hidden/instantiate_tensor.o
In file included from ../../../boost/numeric/ublas/tensor/operators_arithmetic.hpp:16,
                 from ../../../boost/numeric/ublas/tensor.hpp:17,
                 from examples/tensor/instantiate_tensor.cpp:13:
../../../boost/numeric/ublas/tensor/expression_evaluation.hpp: In instantiation of 'constexpr auto boost::numeric::ublas::detail::all_extents_equal(const boost::numeric::ublas::detail::unary_tensor_expression<T, E, OP>&, boost::numeric::ublas::extents<es ...>&) [with T = boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >; E = boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >; OP = operator*<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > >, boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > > >(boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >::const_reference, const boost::numeric::ublas::detail::tensor_expression<boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >, boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > > >&)::<lambda(const auto:50&)>; long unsigned int ...es = {3}; boost::numeric::ublas::extents<es ...> = boost::numeric::ublas::extents_core<long unsigned int, 3>]':
../../../boost/numeric/ublas/tensor/expression_evaluation.hpp:193:30:   required from 'constexpr auto boost::numeric::ublas::detail::all_extents_equal(const boost::numeric::ublas::detail::tensor_expression<T, EL>&, boost::numeric::ublas::extents<es ...>&) [with T = boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >; D = boost::numeric::ublas::detail::unary_tensor_expression<boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >, boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >, operator*<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > >, boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > > >(boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >::const_reference, const boost::numeric::ublas::detail::tensor_expression<boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >, boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > > >&)::<lambda(const auto:50&)> >; long unsigned int ...es = {3}; boost::numeric::ublas::extents<es ...> = boost::numeric::ublas::extents_core<long unsigned int, 3>]'
../../../boost/numeric/ublas/tensor/expression_evaluation.hpp:267:35:   required from 'void boost::numeric::ublas::detail::eval(tensor_type&, const boost::numeric::ublas::detail::tensor_expression<T, EL>&) [with tensor_type = boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >; derived_type = boost::numeric::ublas::detail::unary_tensor_expression<boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >, boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >, operator*<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > >, boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > > >(boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >::const_reference, const boost::numeric::ublas::detail::tensor_expression<boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >, boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > > >&)::<lambda(const auto:50&)> >]'
../../../boost/numeric/ublas/tensor/function/../tensor/tensor_static_rank.hpp:166:17:   required from 'boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, e>, L, std::vector<T> > >::tensor_core(const boost::numeric::ublas::detail::tensor_expression<T, D>&) [with T = boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >; D = boost::numeric::ublas::detail::unary_tensor_expression<boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >, boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >, operator*<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > >, boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > > >(boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >::const_reference, const boost::numeric::ublas::detail::tensor_expression<boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >, boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > > >&)::<lambda(const auto:50&)> >; V = float; L = boost::numeric::ublas::basic_column_major<>; long unsigned int N = 3]'
examples/tensor/instantiate_tensor.cpp:64:30:   required from here
../../../boost/numeric/ublas/tensor/expression_evaluation.hpp:240:12: sorry, unimplemented: unexpected AST of kind nontype_argument_pack
  240 |       if(e != expr.e.extents())
      |          ~~^~~~~~~~~~~~~~~~~~~
../../../boost/numeric/ublas/tensor/expression_evaluation.hpp:240: confused by earlier errors, bailing out

    "gcc-11"   -fvisibility-inlines-hidden -std=c++20 -fPIC -m64 -O3 -finline-functions -Wno-inline -Wall -fvisibility=hidden -O3 -std=c++20 -Wall -pedantic -Wextra -Wno-unknown-pragmas -DBOOST_ALL_NO_LIB=1 -DBOOST_UBLAS_NO_EXCEPTIONS -DNDEBUG  -I"../../.."  -c -o "../../../bin.v2/libs/numeric/ublas/examples/tensor/gcc-11/release/threading-multi/visibility-hidden/instantiate_tensor.o" "examples/tensor/instantiate_tensor.cpp"

...failed gcc.compile.c++ ../../../bin.v2/libs/numeric/ublas/examples/tensor/gcc-11/release/threading-multi/visibility-hidden/instantiate_tensor.o...
...skipped <p../../../bin.v2/libs/numeric/ublas/examples/tensor/gcc-11/release/threading-multi/visibility-hidden>instantiate_tensor for lack of <p../../../bin.v2/libs/numeric/ublas/examples/tensor/gcc-11/release/threading-multi/visibility-hidden>instantiate_tensor.o...
gcc.compile.c++ ../../../bin.v2/libs/numeric/ublas/examples/tensor/gcc-11/release/threading-multi/visibility-hidden/multiply_tensors_product_function.o
In file included from ../../../boost/numeric/ublas/tensor/operators_arithmetic.hpp:16,
                 from ../../../boost/numeric/ublas/tensor.hpp:17,
                 from examples/tensor/multiply_tensors_product_function.cpp:14:
../../../boost/numeric/ublas/tensor/expression_evaluation.hpp: In instantiation of 'constexpr auto boost::numeric::ublas::detail::all_extents_equal(const boost::numeric::ublas::detail::unary_tensor_expression<T, E, OP>&, boost::numeric::ublas::extents<es ...>&) [with T = boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >; E = boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >; OP = operator*<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > >, boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > > >(boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >::const_reference, const boost::numeric::ublas::detail::tensor_expression<boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >, boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > > >&)::<lambda(const auto:50&)>; long unsigned int ...es = {3}; boost::numeric::ublas::extents<es ...> = boost::numeric::ublas::extents_core<long unsigned int, 3>]':
../../../boost/numeric/ublas/tensor/expression_evaluation.hpp:193:30:   required from 'constexpr auto boost::numeric::ublas::detail::all_extents_equal(const boost::numeric::ublas::detail::tensor_expression<T, EL>&, boost::numeric::ublas::extents<es ...>&) [with T = boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >; D = boost::numeric::ublas::detail::unary_tensor_expression<boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >, boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >, operator*<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > >, boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > > >(boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >::const_reference, const boost::numeric::ublas::detail::tensor_expression<boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >, boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > > >&)::<lambda(const auto:50&)> >; long unsigned int ...es = {3}; boost::numeric::ublas::extents<es ...> = boost::numeric::ublas::extents_core<long unsigned int, 3>]'
../../../boost/numeric/ublas/tensor/expression_evaluation.hpp:267:35:   required from 'void boost::numeric::ublas::detail::eval(tensor_type&, const boost::numeric::ublas::detail::tensor_expression<T, EL>&) [with tensor_type = boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >; derived_type = boost::numeric::ublas::detail::unary_tensor_expression<boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >, boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >, operator*<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > >, boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > > >(boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >::const_reference, const boost::numeric::ublas::detail::tensor_expression<boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >, boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > > >&)::<lambda(const auto:50&)> >]'
../../../boost/numeric/ublas/tensor/function/../tensor/tensor_static_rank.hpp:166:17:   required from 'boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, e>, L, std::vector<T> > >::tensor_core(const boost::numeric::ublas::detail::tensor_expression<T, D>&) [with T = boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >; D = boost::numeric::ublas::detail::unary_tensor_expression<boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >, boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >, operator*<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > >, boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > > >(boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >::const_reference, const boost::numeric::ublas::detail::tensor_expression<boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > >, boost::numeric::ublas::tensor_core<boost::numeric::ublas::tensor_engine<boost::numeric::ublas::extents_core<long unsigned int, 3>, boost::numeric::ublas::basic_column_major<>, std::vector<float, std::allocator<float> > > > >&)::<lambda(const auto:50&)> >; V = float; L = boost::numeric::ublas::basic_column_major<>; long unsigned int N = 3]'
examples/tensor/multiply_tensors_product_function.cpp:220:29:   required from here
../../../boost/numeric/ublas/tensor/expression_evaluation.hpp:240:12: sorry, unimplemented: unexpected AST of kind nontype_argument_pack
  240 |       if(e != expr.e.extents())
      |          ~~^~~~~~~~~~~~~~~~~~~
../../../boost/numeric/ublas/tensor/expression_evaluation.hpp:240: confused by earlier errors, bailing out

    "gcc-11"   -fvisibility-inlines-hidden -std=c++20 -fPIC -m64 -O3 -finline-functions -Wno-inline -Wall -fvisibility=hidden -O3 -std=c++20 -Wall -pedantic -Wextra -Wno-unknown-pragmas -DBOOST_ALL_NO_LIB=1 -DBOOST_UBLAS_NO_EXCEPTIONS -DNDEBUG  -I"../../.."  -c -o "../../../bin.v2/libs/numeric/ublas/examples/tensor/gcc-11/release/threading-multi/visibility-hidden/multiply_tensors_product_function.o" "examples/tensor/multiply_tensors_product_function.cpp"

...failed gcc.compile.c++ ../../../bin.v2/libs/numeric/ublas/examples/tensor/gcc-11/release/threading-multi/visibility-hidden/multiply_tensors_product_function.o...
...skipped <p../../../bin.v2/libs/numeric/ublas/examples/tensor/gcc-11/release/threading-multi/visibility-hidden>multiply_tensors_product_function for lack of <p../../../bin.v2/libs/numeric/ublas/examples/tensor/gcc-11/release/threading-multi/visibility-hidden>multiply_tensors_product_function.o...
...failed updating 2 targets...
...skipped 2 targets...

**command :** 
$../../../b2 -j8 release toolset=gcc cxxflags="-O3 -std=c++20" test/tensor
Performing configuration checks

b2 -j8 release  cxxflags="-O3 -std=c++20" test/tensor
Performing configuration checks

    - default address-model    : 64-bit (cached) [1]
    - default architecture     : x86 (cached) [1]
    - symlinks supported       : yes (cached)
    - cxx17_if_constexpr       : yes (cached) [2]
    - cxx17_if_constexpr       : yes (cached) [2]
    - BOOST_COMP_GNUC >= 4.3.0 : yes (cached) [2]

[1] gcc-11
[2] gcc-11/release/python-2.7/threadapi-pthread/threading-multi/visibility-hidden
...patience...
...patience...
...found 3459 targets...
...updating 18 targets...
gcc.compile.c++ ../../../bin.v2/libs/numeric/ublas/test/tensor/test_tensor.test/gcc-11/release/threading-multi/visibility-hidden/extents/test_extents_comparision.o
In file included from ../../../boost/test/test_tools.hpp:45,
                 from ../../../boost/test/unit_test.hpp:18,
                 from test/tensor/extents/test_extents_comparision.cpp:14:
test/tensor/extents/test_extents_comparision.cpp: In member function 'void test_extents_compare::test_extents_static_rank::test_method()':
test/tensor/extents/test_extents_comparision.cpp:96:27: sorry, unimplemented: unexpected AST of kind nontype_argument_pack
   96 |         BOOST_CHECK( n == n );
      |                           ^
../../../boost/test/tools/old/interface.hpp:51:47: note: in definition of macro 'BOOST_TEST_TOOL_PASS_PRED2'
   51 | #define BOOST_TEST_TOOL_PASS_PRED2( P, ARGS ) P
      |                                               ^
../../../boost/test/tools/old/interface.hpp:82:45: note: in expansion of macro 'BOOST_TEST_TOOL_IMPL'
   82 | #define BOOST_CHECK( P )                    BOOST_TEST_TOOL_IMPL( 2, \
      |                                             ^~~~~~~~~~~~~~~~~~~~
test/tensor/extents/test_extents_comparision.cpp:96:9: note: in expansion of macro 'BOOST_CHECK'
   96 |         BOOST_CHECK( n == n );

.......
r<long unsigned int, long int> >, std::pair<float, boost::numeric::ublas::basic_row_major<long unsigned int, long int> >}; typename std::enable_if<(! boost::mpl::is_sequence<C<parameter_pack ...> >::value)>::type = void]'
../../../boost/test/tree/test_case_template.hpp:199:23:   required from 'boost::unit_test::ut_detail::template_test_case_gen<TestCaseTemplate, C<parameter_pack ...>, typename std::enable_if<(! boost::mpl::is_sequence<C<parameter_pack ...> >::value)>::type>::template_test_case_gen(boost::unit_test::const_string, boost::unit_test::const_string, std::size_t) [with TestCaseTemplate = test_functions_real_imag_conj::test_tensor_static_rank_invoker; C = std::tuple; parameter_pack = {std::pair<int, boost::numeric::ublas::basic_column_major<long unsigned int, long int> >, std::pair<int, boost::numeric::ublas::basic_row_major<long unsigned int, long int> >, std::pair<float, boost::numeric::ublas::basic_column_major<long unsigned int, long int> >, std::pair<float, boost::numeric::ublas::basic_row_major<long unsigned int, long int> >}; typename std::enable_if<(! boost::mpl::is_sequence<C<parameter_pack ...> >::value)>::type = void; boost::unit_test::const_string = boost::unit_test::basic_cstring<const char>; std::size_t = long unsigned int]'
test/tensor/functions/test_functions_real_imag_conj.cpp:119:1:   required from here
../../../boost/numeric/ublas/tensor/expression_evaluation.hpp:211:12: sorry, unimplemented: unexpected AST of kind nontype_argument_pack
  211 |       if(e !=  expr.el.extents())
      |          ~~^~~~~~~~~~~~~~~~~~~~~
../../../boost/numeric/ublas/tensor/expression_evaluation.hpp:211: confused by earlier errors, bailing out

    "gcc-11"   -fvisibility-inlines-hidden -std=c++20 -fPIC -m64 -O3 -finline-functions -Wno-inline -Wall -fvisibility=hidden -O3 -std=c++20 -Wall -Wno-unknown-pragmas -Wall -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-but-set-variable -DBOOST_ALL_NO_LIB=1 -DBOOST_TEST_DYN_LINK=1 -DBOOST_TEST_NO_AUTO_LINK=1 -DBOOST_UBLAS_NO_EXCEPTIONS -DNDEBUG  -I"../../.."  -c -o "../../../bin.v2/libs/numeric/ublas/test/tensor/test_tensor.test/gcc-11/release/threading-multi/visibility-hidden/functions/test_functions_real_imag_conj.o" "test/tensor/functions/test_functions_real_imag_conj.cpp"

...failed gcc.compile.c++ ../../../bin.v2/libs/numeric/ublas/test/tensor/test_tensor.test/gcc-11/release/threading-multi/visibility-hidden/functions/test_functions_real_imag_conj.o...
...skipped <p../../../bin.v2/libs/numeric/ublas/test/tensor/test_tensor.test/gcc-11/release/threading-multi/visibility-hidden>test_tensor for lack of <p../../../bin.v2/libs/numeric/ublas/test/tensor/test_tensor.test/gcc-11/release/threading-multi/visibility-hidden>extents/test_extents_comparision.o...
...skipped <p../../../bin.v2/libs/numeric/ublas/test/tensor/test_tensor.test/gcc-11/release/threading-multi/visibility-hidden>test_tensor.run for lack of <p../../../bin.v2/libs/numeric/ublas/test/tensor/test_tensor.test/gcc-11/release/threading-multi/visibility-hidden>test_tensor...
...failed updating 15 targets...
...skipped 3 targets...

**Comment by amitsingh19975** [comment, marked as answer]
If you're using macOS, I recommend using Clang as GCC from homebrew has problems. If you want to use GCC, I recommend building it yourself from the source, which will be very difficult. Even mac alias Clang as GCC by default, and it's pretty involved if you're going to use GCC because of the nature of the OS.

`user-config.jam `
```jam
using clang : : clang++ : <cxxflags>"-std=c++20" <linkflags>"-lboost_unit_test_framework" ;
```

**Reply by zhanggiene** [reply]
I followed the instructions, but clang seems to have problems with c++17 standard.

$../../../b2 -j8 release  cxxflags="-O3" examples/tensor
Performing configuration checks

    - default address-model    : 64-bit (cached) [1]
    - default architecture     : x86 (cached) [1]
    - symlinks supported       : yes (cached)
    - cxx17_if_constexpr       : no  (cached) [2]
    - cxx17_if_constexpr       : no  (cached) [3]

[1] clang-12
[2] clang-darwin-12/release/python-2.7/threadapi-pthread/threading-multi/visibility-hidden
[3] clang-darwin-12/release/build-no/python-2.7/threadapi-pthread/threading-multi/visibility-hidden
...found 1 target...

$more ~/user-config.jam 
using clang : : clang++ : <cxxflags>"-std=c++20" <linkflags>"-lboost_unit_test_framework" ;

**Reply by amitsingh19975** [reply]
We use C++ 20 for tensor, and I would also recommend using `boost.build` to give flags rather than directly using them, such as `<cxxflags>`, `<linkflags>`, etc., in the config file.

**Reply by zhanggiene** [reply]
Thanks! you are correct, parameter should be in the config file only. 
It works when i take out the command line parameters.
Here is the correct command and the correct output, for future reference. I will update the wiki to make it clearer. 
$../../../b2 -j8 variant=debug  examples/tensor
Performing configuration checks

    - default address-model    : 64-bit (cached) [1]
    - default architecture     : x86 (cached) [1]
    - symlinks supported       : yes (cached)
    - cxx17_if_constexpr       : yes (cached) [2]
    - cxx17_if_constexpr       : yes (cached) [2]

[1] clang-12
[2] clang-darwin-12/debug/python-2.7/threadapi-pthread/threading-multi/visibility-hidden
...patience...
...patience...
...found 2766 targets...

**Reply by amitsingh19975** [reply]
You can use the command line to pass the flags, but the way is a bit different. See the `boost.build` document on how to use them. The reason you failed was that you're not using the correct flag to pass `release`. It should be `variant=release`.
